### PR TITLE
Merge branch branch-24.02 into main [skip ci]

### PR DESCRIPTION
**Pull request by NvTimLiu**
Merge branch 'branch-24.02' into main

Change version to 24.02.0

Note: please merge this change with "Create a merge commit"

### CODE REVIEW

**Comment by NvTimLiu** [comment]
This PR is not the final merge to main change.

Draft the PR for the in-advance review purpose

NOTE: There are still some open PRs targeting 24.02, will do the final merge after they get merged.

https://github.com/NVIDIA/spark-rapids/pulls?q=is%3Apr+is%3Aopen+base%3Abranch-24.02+

File: CHANGELOG.md [hunk 1]
```diff
@@ -125,7 +125,7 @@ Generated on 2024-01-31
 |[#10123](https://github.com/NVIDIA/spark-rapids/pull/10123)|Change version to v23.12.1 [skip ci]|
 |[#10122](https://github.com/NVIDIA/spark-rapids/pull/10122)|Init changelog for v23.12.1 [skip ci]|
 |[#10121](https://github.com/NVIDIA/spark-rapids/pull/10121)|[DOC] update download page for db hot fix  [skip ci]|
-|[#10116](https://github.com/NVIDIA/spark-rapids/pull/10116)|Upgrade to 23.12.1|
+|[#10116](https://github.com/NVIDIA/spark-rapids/pull/10116)|Upgrade to 23.12.1-SNAPSHOT|
```
Comment (NvTimLiu) [inline CHANGELOG.md:128]: This change is to fix my typo, detailed discussion:  https://github.com/NVIDIA/spark-rapids/pull/10383#discussion_r1479994296

**Review by jlowe — APPROVED** [review]

File: .github/workflows/blossom-ci.yml [hunk 2]
```diff
@@ -55,6 +55,7 @@ jobs:
       tgravescs,\
       wbo4958,\
       wjxiz1992,\
+      sperlingxx,\
       pxLi,\
```
[the remark this answers is not in the document]
Comment (jlowe) [reply .github/workflows/blossom-ci.yml:59]: Yes, originates from #9959 which went into branch-24.02 and thus part of this merge PR.

[the remark this answers is not in the document]
Comment (NVnavkumar) [reply .github/workflows/blossom-ci.yml:59]: yeah, I understand that. Just noticed that pxLi wasn't removed.

[the remark this answers is not in the document]
Comment (sameerz) [reply .github/workflows/blossom-ci.yml:59]: Put up https://github.com/NVIDIA/spark-rapids/pull/10396 to address Navin's comment in 24.04.

**Comment by NvTimLiu** [comment]
build

**Review by GaryShen2008 — APPROVED** [review]

**Pull request merged**